mlir/include/mlir/Dialect/GPU/Transforms/Passes.h

```diff
@@ -15,6 +15,7 @@
 
 #include "Utils.h"
 #include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/SPIRV/IR/SPIRVAttributes.h"
```

Comment:
This feels off: you have GPU dialect transforms depending on SPIRV attributes/dialect (esp in header). Why is this pass in GPU dialect rather than SPIRV one?

Reply:
Back when the compilation redesign was happening we decided to add the attach* passes in GPU to avoid polluting lower level dialects with GPU includes. However, I do agree that include shouldn't be there, as far as I could tell, that include is only needed by one pass option `mlir::spirv::TargetEnvAttr::kUnknownDeviceID`, so it should be possible to remove it. I'll fix it.

Reply:
SG. Thanks @jpienaar & @fabianmcg!